mailroom.py

```diff
@@ -0,0 +1,275 @@
+#   $Id: play01.py,v 1.11 2017/02/12 20:27:25 larry Exp $
+
+# Various import
+# ---------------------------------------------------------------
+import sys
+import pickle
+import os.path
+import datetime
+# ---------------------------------------------------------------
+
+# Create dictionary
+donors = {
+"Bugs Bunny":[1000, 2000, 3000],
+"Elmer Fudd":[500, 1200, 1600],
+"Foghorn Leghorn":[800, 1600, 2400],
+"Daffy Duck":[550, 750, 2400],
+"Johnny Quest":[300, 600, 900],
+"Marvin Martian":[5000, 10000, 20000],
+"Porky Pig":[2500, 2800, 3000]}
+
+# Open the pickle
+try:
+    # Open the pickle
+    file_obj = open("/home/larry/UW/Play/donors", 'wb')
+
+    # Put the donor dictionary in the pickle
+    pickle.dump(donors, file_obj)
+
+    # Close the pickle
+    file_obj.close()
+
+except:
+    print("Could not open file, system exiting.")
+    sys.exit()
+
+# Open pickle for use, data looks like the following: name + donation amounts
+# {'Bugs Bunny': [1000, 2000, 3000], etc.
+donor_dict = pickle.load(open("donors", "rb"))
+
+class menu_play:
+    """
+    Reads a pickle database into memory
+    Dynamic menu
+    Makes method choice based on user input
+    """
+
+    def __init__(self, msg = ["Placeholder"]):
+        """
+        Parameter is a list of from 1 to 9 strings
+        """
+        self.msg = msg
+
+    def get_input(self):
+        """
+        User selects a routine
+        """
+        # Creates available choices list
+        tst = [str(x) for x in range(1, len(self.msg) + 1)]
+
+        # First element in menu choices
+        v1 = tst[0]
+
+        # Last element in menu choices
+        v2 = tst[-1]
+
+        # Show menu
+        self.make_tui()
+
+        # Insist on correct input & give feedback if wrong
+        while True:
+            chc = input()
+            if not chc in tst:
+                self.make_tui()
+                print("You must enter a number from %s to %s." % (v1, v2))
+            else:
+                break
+
+        # All available menu methods
+        my_methods = {"1" : self.m1,
+                      "2" : self.m2,
+                      "3" : self.m3,
+                      "4" : self.m4,
+                      "5" : self.m5,
+                      "6" : self.m6,
+                      "7" : self.m7,
+                      "8" : self.m8,
+                      "9" : self.m9}
+
+        # Call menu method based on user choice
+        my_methods[chc]()
+
+    def m1(self):
```

Comment:
Try to use functions names that say what the function does.  m1 doesn't mean anything to another programmer.